src/human_player_module.c

```diff
@@ -83,6 +85,158 @@ bool validate_card(char *entered_value)
     return false;
 }
 
+ /**
+  * @brief  human player processes end turn request
+  *
+  * @return ret_type_e: RET_SUCCESS on success; 
+  *                     RET_FAILURE on fail
+  */
+ret_type_e human_process_end_turn_request(void)
+{
+    ret_type_e ret = RET_FAILURE;
+
+    if (g_card_requested)
+    {
+        g_card_requested = false;
+        end_turn(HUMAN);
+        ret = RET_SUCCESS;
+    }
+    else
+    {
+        printf("!!Warning!! Please draw a card before you can end your turn \n");
+        display_player_deck(HUMAN);
+    }
+
+    return ret;
+}
+
+/**
+ * @brief human player discards a normal card 
+ *
+ * @return ret_type_e: RET_SUCCESS on successful;
+ *                     RET_INVALID_CARD on invalid input
+ *                     RET_INVALID_TURN
+ */
+ret_type_e human_process_normal_card(const char* user_input)
+{
+    if (validate_card(user_input))
+    {
+        //To be added if the format of user input is different
+        Card_t human_card_choice = map_user_input(user_input);
+
+        //Check if the card is from the human player deck
+        if (!is_human_card(&human_card_choice))
+        {
+            printf("!!Warning!! Please select a card from your deck! \n");
+            return RET_INVALID_CARD;
+        }
+
+        //If valid add the card to the discard_pile
+        if (check_is_valid_card(&human_card_choice))
+        {
+            remove_card_from_deck(&(g_players[HUMAN].cards_on_hand), human_card_choice);
+            add_card_at_end(g_discard_pile, human_card_choice);
+            memcpy(&g_card_on_table, &human_card_choice, sizeof(Card_t));
+            end_turn(HUMAN);
+            g_card_requested = false;
+            return RET_SUCCESS;
+        }
+        else
+        {
+            invalid_card_warning();
+            return RET_INVALID_CARD;
+        }
+    }
+    else
+    {
+        invalid_card_warning();
+        return RET_INVALID_CARD;
+    }
+}
+
+/**
+ * @brief human player discards a normal card
```

Comment:
Can you explain more on the brief part here? Seems its similar to the previous function.

Reply:
No features change, just make the code readable by separating them into functions.